File: 3242-count-elements-with-maximum-frequency/3242-count-elements-with-maximum-frequency.cpp
class Solution {
public:
    int maxFrequencyElements(vector<int>& nums) {
        unordered_map<int,int>freq;

        for(int i=0; i<nums.size(); i++){
            freq[nums[i]]++;
        }

        int Maxans = INT_MIN;
        for(auto i:freq){
           if(i.second >Maxans){
                Maxans = i.second;
           }
        }
        
        int count =0;
        for(auto i:freq){
            if(i.second == Maxans){
                count+= Maxans;
            }
        }

        return count;
    }
};

File: 3242-count-elements-with-maximum-frequency/3242-count-elements-with-maximum-frequency.java
class Solution {
    public int maxFrequencyElements(int[] nums) {
        Map<Integer, Integer> freq = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            freq.put(nums[i], freq.getOrDefault(nums[i], 0) + 1);
        }

        int maxAns = Integer.MIN_VALUE;
        for (Map.Entry<Integer, Integer> entry : freq.entrySet()) {
            if (entry.getValue() > maxAns) {
                maxAns = entry.getValue();
            }
        }

        int count = 0;
        for (Map.Entry<Integer, Integer> entry : freq.entrySet()) {
            if (entry.getValue() == maxAns) {
                count += maxAns;
            }
        }

        return count;

    }
}